### ISSUE TITLE
Blueprint table elements do not use correct roles

### ISSUE THREAD

**Issue by styu** [palantir/blueprint#6224]
You can use VoiceOver on Mac on https://blueprintjs.com/docs/#table/features to trivially see this - the elements don't correctly declare e.g. table, row, cell, etc. roles so that the screenreader is unable to correctly read each row. Instead, each cell gets read one at a time
